### DISCUSSION TITLE
Add image to checkbox widget

### DISCUSSION THREAD

**Discussion by ghazalbangash** [pyvista/pyvista#2880] [Q&A] [answered]
Hi there.Is it possible to add an image instead of a color to add_checkbox_button_widget.I know vtk supports this by using  vtk.vtkPNGReader() and then setting the buttonTexture as the image. but I dont have any access to the inner variables of the function. Thank you for your time.

**Comment by banesullivan** [comment, marked as answer]
Perhaps you can set this after creation:

```py
import pyvista as pv
from pyvista import examples

tex = examples.download_puppy_texture()

mesh = pv.Sphere()

p = pv.Plotter(notebook=0)
actor = p.add_mesh(mesh)


def toggle_vis(flag):
    actor.SetVisibility(flag)


widget = p.add_checkbox_button_widget(toggle_vis, value=True)

rep = widget.GetRepresentation()
rep.SetButtonTexture(1, tex.to_image())

p.show()
```

https://user-images.githubusercontent.com/22067021/176029190-5594f96e-9989-46c3-90d5-0138493bbc8e.mov




We could override the `color_on` and `color_off` options to accept `vtkImageData`/`pyvista.UniformGrid`s, `vtkTexture`s, etc.

**Comment by ghazalbangash** [comment]
Thanks a lot , that's exactly what I needed.
